lighthouse-cli/printer.ts

```diff
@@ -17,45 +17,69 @@
 
 'use strict';
 
-const fs = require('fs');
-const ReportGenerator = require('../lighthouse-core/report/report-generator');
-const Formatter = require('../lighthouse-core/formatters/formatter');
-const log = require('../lighthouse-core/lib/log');
-
 /**
  * An enumeration of acceptable output modes:
- * <ul>
- *   <li>'pretty': Pretty print the results</li>
- *   <li>'json': JSON formatted results</li>
- *   <li>'html': An HTML report</li>
- * </ul>
- * @enum {string}
+ *   'pretty': Pretty print the results
+ *   'json': JSON formatted results
+ *   'html': An HTML report
  */
-const OUTPUT_MODE = {
-  pretty: 'pretty',
-  json: 'json',
-  html: 'html'
+enum OutputMode { pretty, json, html };
+type Mode = 'pretty' | 'json' | 'html';
+const Modes = ['pretty', 'json', 'html'];
```

Comment:
We allow users to pass one of a know `Mode` for output so we need a Mode type to limit that.
We also provide a known list of safe string values `Modes`

We then need to index into our `Modes` list. Instead of doing that with a raw int value we use an enum for more readable code.